### PR TITLE
Test the presence of OptionsResolver when the body_converter validation option is enabled

### PR DESCRIPTION
**Pull request by 2binfree**

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -110,7 +110,20 @@ public function getConfigTreeBuilder()
                 ->arrayNode('body_converter')
                     ->canBeEnabled()
                     ->children()
-                        ->scalarNode('validate')->defaultFalse()->end()
+                        ->scalarNode('validate')
+                            ->defaultFalse()
+                            ->beforeNormalization()
+                                ->ifTrue()
+                                ->then(function() {
+                                    if (!class_exists('Symfony\Component\OptionsResolver\OptionsResolver')) {
```
Comment (xabbuh) [inline DependencyInjection/Configuration.php]: let's use the `class` constant here

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -110,7 +110,20 @@ public function getConfigTreeBuilder()
...
+                                        throw new InvalidConfigurationException(
```
Comment (xabbuh) [inline DependencyInjection/Configuration.php]: please just place this on a single line

Comment (2binfree) [reply DependencyInjection/Configuration.php]: Ok but the line is going to be very long.
[There MUST NOT be a hard limit on line length; the soft limit MUST be 120
characters; lines SHOULD be 80 characters or less.](https://www.php-fig.org/psr/psr-2/)

Comment (GuilhemN) [reply DependencyInjection/Configuration.php]: We consider exception messages as exceptions :)

File: DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -110,7 +111,17 @@ public function getConfigTreeBuilder()
...
                     ->canBeEnabled()
                     ->children()
-                        ->scalarNode('validate')->defaultFalse()->end()
+                        ->scalarNode('validate')
+                            ->defaultFalse()
+                            ->beforeNormalization()
+                                ->ifTrue()
+                                ->then(function () {
+                                    if (!class_exists(OptionsResolver::class)) {
+                                        throw new InvalidConfigurationException("'body_converter.validate: true' require OptionsResolver component installation ( composer require symfony/options-resolver )");
```
Comment (GuilhemN) [inline DependencyInjection/Configuration.php]: `requires`

**Comment by GuilhemN** [comment]
Thank you @2binfree!

File: DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -110,7 +111,17 @@ public function getConfigTreeBuilder()
                 ->arrayNode('body_converter')
                     ->canBeEnabled()
                     ->children()
-                        ->scalarNode('validate')->defaultFalse()->end()
+                        ->scalarNode('validate')
+                            ->defaultFalse()
+                            ->beforeNormalization()
+                                ->ifTrue()
```
Comment (xabbuh) [inline DependencyInjection/Configuration.php]: should be `always()` (see #1918)

**Comment by goetas** [comment]
@GuilhemN can I ask why this was merged if it was breaking the test suite?

**Comment by GuilhemN** [comment]
@goetas oops, I thought all test failures were unrelated (I didn't check all cases 😶)... It proves we need to fix our test suite :)

**Pull request merged**